### ISSUE TITLE
Revision Upgrades Break Validation Webhook

### ISSUE THREAD

**Issue by therealmitchconnors** [istio/istio#28880]
**Bug description**
Upgrading from a default revision (istiod) to a new revision (istiod-foo) may result in all istio config becoming unchangeable, blocking upgrades and downgrades, after the original revision (istiod) is removed.

[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
Validation moved to new revision OR Validation failurePolicy = Ignore.

**Steps to reproduce the bug**
```
./bin/istioctl install
./bin/istioctl install --revision foo
// wait for ValidatingWebhookConfiguration failurePolicy=Fail, usually less than a minute
kubectl delete service -n istio-system istiod
kubectl apply -f samples/bookinfo/networking/virtual-service-all-v1.yaml 
// any istio resources will fail to modify
```

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**
Confirmed in 1.7.4.  Suspected in all versions since 1.6

**Comment by therealmitchconnors** [comment]
Oddly, deleting the Istiod Deployment instead of the Service causes the webhook to be reset to failurePolicy:Ignore, which is a better, but still broken state.

**Comment by mandarjog** [comment]
@therealmitchconnors thanks. Can you clarify upgrade `from` and `to` versions?

**Comment by therealmitchconnors** [comment]
I have reproduced from 1.7.4 to 1.7.4.  Others are suspected as well.

**Comment by howardjohn** [comment]
This is the same root cause of https://github.com/istio/istio/issues/27501

https://docs.google.com/document/d/1k9V3D5jhZdFvEdJhw-T7mHCEIjkK0hFSdvtJIWEqIs0 is the design for the fix

**Comment by therealmitchconnors** [comment]
FWIW, I think we can mitigate this bug for 1.8 with changes to docs.

**Comment by christian-posta** [comment]
Shouldn't this be a release blocker @jacob-delgado ?

**Comment by christian-posta** [comment]
I understand the TOC voted this not a release blocker which is unfortunate. This functionality being broken prevents canary upgrades. Unless there are some know workarounds? or order of operations to prevent getting into this state?

**Comment by howardjohn** [comment]
The general policy for release blocker is it is a regression from previous
versions. As this has been broken since ~1.6, it is not a release blocker
per that guidance. However, I think we all agree its a big issue and are
working to address it. The workaround in the meantime is to change the
Validating webhook address to the new revision, or make a `istiod` Service
selecting the new revision.

**Comment by thorion3006** [comment]
The issue is still present in v1.9.1. To get validations to work with revisions, I add an extra service called istiod and point it to the current revision.
```yaml
apiVersion: v1
kind: Service
metadata:
  name: istiod
  namespace: istio-system
spec:
  ports:
    - name: grpc-xds
      port: 15010
      protocol: TCP
      targetPort: 15010
    - name: https-dns
      port: 15012
      protocol: TCP
      targetPort: 15012
    - name: https-webhook
      port: 443
      protocol: TCP
      targetPort: 15017
    - name: http-monitoring
      port: 15014
      protocol: TCP
      targetPort: 15014
  selector:
    app: istiod
    istio.io/rev: 1-9-1
```

I just update the selector for this service whenever I switch to a new revision.

**Comment by mitchellmaler** [comment]
Running 1.10 I ran into this same issue. I notice the docs still link to this issue calling it out as a bug.
https://istio.io/latest/docs/setup/upgrade/canary/
Should this issue be re-opened until a fix is created?

It also states we can point the service to the revision. Could that cause issues if we upgrade to a new revision like 1.11 and there is a difference between 1.10 and 1.11 and all 1.10 namespaces will have issues?

**Comment by linsun** [comment]
pretty sure this is already fixed but haven't made into releases older than 1.10.  so if you are upgrading from an older release to 1.10, you will hit it.

cc @Monkeyanator.

**Comment by linsun** [comment]
Note: this call out is removed in 1.11

**Comment by sonnyhcl** [comment]
> pretty sure this is already fixed but haven't (and won't) made into releases older than 1.10. so if you are upgrading from an older release to 1.10, you will hit it.

If the community won't made the fix into releases older than 1.10, should we at least have a warning message in release note/upgrade note?  @linsun
